### PR TITLE
Add pg_stat_statements section

### LINKED ISSUE #832: ABL [Docs RFC] Add section on using `pg_stat_statements` for Cloud Visibility
# Describe change in content, appearance, or functionality

For Cloud docs, we need a section that describes [pg_stat_statements](https://www.postgresql.org/docs/14/pgstatstatements.html).

All Cloud services come with the extension pre-installed.
They need to connect with a Postgres Client (e.g. psql).

In order to filter out any superuser query statistics they need to add the following WHERE clause to their queries:

e.g. 
`SELECT * FROM pg_stat_statements WHERE pg_get_userbyid(userid) = current_user;`

In general, this allows them to view performance statistics of their queries.
E.g. identify long running queries, I/O heavy queries, memory heave queries, etc...

Maybe as a fast follow we could recommend some queries they could run to improve DB performance?

# Subject matter expert (SME)

[If known, who is a good person to ask about this topic]

# Deadline

Anytime

# Any further info

[Anything else you want to add, or further links]

### PR DESCRIPTION
**Pull request by charislam**
# Description

Add section on `pg_stat_statements` to Metrics page:
- `pg_stat_statements` comes pre-installed
- How to query the view, including filtering out superuser queries
- Provide useful sample queries

Build preview: https://docs-dev.timescale.com/docs-charis-pg_stat_statements/cloud/charis-pg_stat_statements/metrics/

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Fixes #832

### CODE REVIEW

**Comment by charislam** [comment]
@lukas-bernert feedback on what I have right now would be helpful, I'm still waiting for example queries which I was going to take from the blog post draft, unless you have another source?

**Comment by lukas-bernert** [comment]
@charislam the docs so far look good 👍 
Ryan is still working on the blogpost draft. The majority should be finished today.
https://docs.google.com/document/d/1br8gEHDRx0ZCX2GOlOBRAgfGiIsXwcjwts7NdqJL4OA/edit#heading=h.bb2uudl5rr0u

**Review by lukas-bernert — APPROVED** [review]

File: cloud/metrics.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
 # Metrics
+You can view your service metrics from Timescale Cloud's [metrics
+dashboard](#metrics-dashboard). This dashboard gives you service-level
+information, such as CPU, memory, and storage usage.
+
+You can also view your query statistics by using the pre-installed
```
Comment (Rajakavitha1) [inline cloud/metrics.md:6]:
```suggestion
You can also view your query-level statistics by using the pre-installed
```

File: cloud/metrics.md [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+[`pg_stat_statements`](#query-level-statistics-with-pg-stat-statements)
+extension from a PostgreSQL client.
+
+## Metrics dashboard
 Timescale Cloud provides a Metrics dashboard for managing your services. You can
```
Comment (Rajakavitha1) [inline cloud/metrics.md:11]:
```suggestion
Timescale Cloud provides a metrics dashboard for managing your services. You can
```

File: cloud/metrics.md [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
 see the Metrics dashboard in your Timescale Cloud account by navigating to the
```
Comment (Rajakavitha1) [inline cloud/metrics.md:12]:
```suggestion
see the metrics dashboard in your Timescale Cloud account by navigating to the
```

File: cloud/metrics.md [hunk 4]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+statements. It comes pre-installed with Timescale Cloud.
+
+<highlight type="note">
+For more information about `pg_stat_statements`, see the 
+[PostgreSQL documentation](https://www.postgresql.org/docs/current/pgstatstatements.html).
+</highlight>
+
+## Query the pg_stat_statements view
+You can view statistics for your queries through the `pg_stat_statements`
+extension, which provides a `pg_stat_statements` view. The recorded statistics
```
Comment (Rajakavitha1) [inline cloud/metrics.md:47]:
```suggestion
extension. The recorded statistics
```

File: cloud/metrics.md [hunk 5]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+include the time spent planning and executing each query; the number of blocks
+hit, read, and written; and more.
+
+You can query the `pg_stat_statements` view as you would any PostgreSQL view.
```
Comment (Rajakavitha1) [inline cloud/metrics.md:51]:
```suggestion
You can query the `pg_stat_statements` view as you would in any PostgreSQL view.
```

File: cloud/metrics.md [hunk 6]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+The full view includes superuser queries, which are used by Timescale Cloud to
+manage your service in the background. To limit the results to only your
```
Comment (Rajakavitha1) [inline cloud/metrics.md:53]:
```suggestion
manage your service in the background. To view only your
```

File: cloud/metrics.md [hunk 7]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+queries, filter by the current user.
+
+Connect to your database using a PostgreSQL client, such as [`psql`][psql], and
+run:
+```sql
+SELECT * FROM pg_stat_statements WHERE pg_get_userbyid(userid) = current_user;
+```
+
+<highlight type="important">
```
Comment (Rajakavitha1) [inline cloud/metrics.md:62]: May be this can go in the begining para of ## Query the pg_stat_statements view.

File: cloud/metrics.md [hunk 8]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+You cannot currently enable `track_io_timing` for your database. Statistics that
+depend on `track_io_timing`, such as `blk_read_time` and `blk_write_time`, are
+not collected.
+</highlight>
+
+## Example usage
+With `pg_stat_statements`, you can view performance statistics to help you
```
Comment (Rajakavitha1) [inline cloud/metrics.md:69]:
```suggestion
With `pg_stat_statements`, you can view performance statistics that help you
```

File: cloud/metrics.md [hunk 9]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+monitor and optimize your queries.
+
+Here are some sample scenarios to try.
+
+### Identify long-running queries
```
Comment (Rajakavitha1) [inline cloud/metrics.md:74]:
```suggestion
### Identifying long-running queries
```

File: cloud/metrics.md [hunk 10]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+SELECT calls,
+    mean_exec_time,
+    query
+FROM pg_stat_statements
+WHERE pg_get_userbyid(userid) = current_user
+ORDER BY mean_exec_time DESC
+LIMIT 5;
+```
+
+### Identify queries with highly variable execution time
```
Comment (Rajakavitha1) [inline cloud/metrics.md:86]:
```suggestion
### Identifying queries with highly variable execution time
```

File: cloud/metrics.md [hunk 11]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+</highlight>
+
+## Example usage
+With `pg_stat_statements`, you can view performance statistics to help you
+monitor and optimize your queries.
+
+Here are some sample scenarios to try.
+
+### Identify long-running queries
+Identify the 5 longest-running queries by their mean execution time:
```
Comment (Rajakavitha1) [inline cloud/metrics.md:75]:
```suggestion
To identify the five longest-running queries by their mean execution time:
```

**Review by Rajakavitha1 — APPROVED** [review]

File: cloud/metrics.md [hunk 12]
```diff
@@ -23,3 +32,73 @@ In some cases, gray vertical bars display on the metrics dashboard, like this:
...
+# Query-level statistics with pg_stat_statements
+The `pg_stat_statements` extension gives you query-level statistics for your SQL
+statements. It comes pre-installed with Timescale Cloud.
+
+<highlight type="note">
+For more information about `pg_stat_statements`, see the 
+[PostgreSQL documentation](https://www.postgresql.org/docs/current/pgstatstatements.html).
+</highlight>
+
+<highlight type="note">
```
Comment (Loquacity) [inline cloud/metrics.md:45]:
```suggestion
<highlight type="important">
```

**Review by Loquacity — APPROVED** [review]
One non-blocking comment. How close to being ready to merge is this?

**Comment by charislam** [comment]
> @charislam do you think **https://docs.timescale.com/cloud/latest/metrics/#query-level-statistics-with-pg-stat-statements** will be the final URL once released?

@lukas-bernert Depends whether the Cloud reorg is merged (I believe the plan is to release it soon @Loquacity?) After the reorg, I think it would be **https://docs.timescale.com/cloud/latest/service-metrics/#query-level-statistics-with-pg-stat-statements**

**Comment by lukas-bernert** [comment]
Nevermind on the above. 
We'll use shortlinks which we can update later on.
https://iobeam.slack.com/archives/C034ZM87486/p1647881866928009?thread_ts=1647873036.349759&cid=C034ZM87486

Thanks @charislam for updating 🔥

**Pull request merged**